BrainPortal/app/models/data_provider.rb

```diff
@@ -999,6 +999,12 @@ def provider_full_path(userfile)
   def self.pretty_type #:nodoc:
     self.to_s
   end
+  
+  # this is a method that will return category type for this data_provider
```

Comment:
Maybe returning nil in here, and explaining that returning nil is the convention to HIDE a data provider class from the interface, so that other programmers know this feature is available?